subsys/fs/Kconfig.fatfs

```diff
@@ -88,6 +88,38 @@ config FS_FATFS_NUM_DIRS
 	  at compile-time.
 	  This affects use of fs_opendir on FAT type mounted file systems.
 
+config FS_FATFS_HAS_RTC
+	bool "Timestamping support"
+	help
+	  Enable file system timestamping instead of using a hardcoded date
+	  for all operations. Requires an application supplied implementation
+	  of `get_fattime`. Format of the uint32_t bits are as follows:
+	    31:25 = Year from 1980
+		24:21 = Month (1..12)
+		20:16 = Day of month (1..31)
+	    15:11 = Hour (0..23)
+		10: 5 = Minute (0..59)
+		 4: 0 = Seconds/2 (0..29)
+
+config FS_FATFS_USE_LABEL
+	bool "Set/Get filesystem labels"
+	help
+	  Enable retrieving and setting the filesystem labels through
+	  `f_getlabel` and `f_setlabel`.
+
+config FS_FATFS_USE_EXPAND
+	bool "Pre-allocate contiguous files"
+	help
+	  Enable pre-allocating a contiguous data area to the file through
+	  `f_expand`. Unlike expansion through `f_truncate`, `f_expand` does
+	  not write to the expanded memory area.
```

Comment:
I think that this should be only available when using RAW access. Also, the option depends on `CONFIG_FATFS_READ_ONLY=n`

Reply:
Added the missing dependency to this symbol and `FS_FATFS_MKFS`.
Enabling this option without the `RAW` symbol I have proposed is not incorrect, although probably not too useful.
Either way the symbol does not exist yet so I cannot add any dependency.